src/test/run-make-fulldeps/coverage-reports-base/Makefile

```diff
@@ -13,15 +13,13 @@
 BASEDIR=../coverage-reports-base
 SOURCEDIR=../coverage
 
-ifeq ($(UNAME),Darwin)
-# FIXME(richkadel): It appears that --debug is not available on MacOS even when not running
-# under CI.
-NO_LLVM_ASSERTIONS=1
-endif
-
 # The `llvm-cov show` flag `--debug`, used to generate the `counters` output files, is only enabled
-# if LLVM assertions are enabled. Some CI builds disable debug assertions.
-ifndef NO_LLVM_ASSERTIONS
+# if LLVM assertions are enabled. Requires Rust config `llvm/optimize` and not
+# `llvm/release_debuginfo`. Note that some CI builds disable debug assertions (by setting
+# `NO_LLVM_ASSERTIONS=1`), so it is not OK to fail the test, but `bless`ed test results cannot be
+# generated without debug assertions.
```

Comment:
I guess we can consider this like tests which only run on certain platforms, except it only runs in certain configurations. It's not ideal, because if someone makes a large change that affects these tests and isn't in the right configuration, they won't be able to update the tests.

That being said, I'm not sure what we can do differently. I wish there was an LLVM build option to enable only this flag.

One mitigation is to add a warning if someone tries to bless and we can't update all the files. (To keep it from being too noisy.. Maybe only print it if they specify bless and one of the other files doesn't match? This kind of logic is difficult in a Makefile which is another reason I think we should move away from those.)